KarhooUISDK/CommonUI/Views/KarhooNewTextField.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  KarhooNewTextField.swift
+//  KarhooUISDK
+//
+//  Created by Bartlomiej Sopala on 16/01/2023.
+//  Copyright © 2023 Flit Technologies Ltd. All rights reserved.
+//
+
+import SwiftUI
+
+struct KarhooNewTextField: View {
+   
+    @Binding var textFieldText: String
+    @Binding var isTextfieldValid: Bool
+    @State private var isFirstResponder: Bool = false
+
+    var placeholder: String
+    var errorMessage: String
+    var contentType: KarhooTextInputViewContentType
+    
+    
+    var body: some View {
+        VStack(alignment: .leading, spacing: 0) {
+            HStack{
+                TextField(placeholder, text: $textFieldText, onEditingChanged: { (editingChanged) in
+                    if editingChanged {
+                        isFirstResponder = true
+                    } else {
+                        isFirstResponder = false
+                    }
+                })
+                .onChange(of: textFieldText) { newValue in
+                    withAnimation {
+                        isTextfieldValid = getTextFieldValidity(newValue: newValue)
+                    }
+                }
+                Button {
+                    textFieldText = ""
+                } label: {
+                    Image(uiImage: .uisdkImage("kh_uisdk_cross_in_circle"))
+                        .frame(
+                            width: UIConstants.Dimension.Icon.standard,
+                            height:UIConstants.Dimension.Icon.standard
+                        )
+                }
+            }
+            .padding(.vertical, UIConstants.Spacing.medium)
+            .padding(.leading, UIConstants.Spacing.standard)
+            .padding(.trailing, UIConstants.Spacing.small)
+            .addBorder(getBorderColor(), cornerRadius: UIConstants.CornerRadius.medium)
+            
+            if !isTextfieldValid {
```

Comment:
I'd change the property to simple `showError`. Keep it simple.

Reply:
This property is available for other elements, like "big/main" view model for all checkout of will be available for passenger details VM. Then if make sense to check, `if isTextfieldValid` rather than `if showError`. So I prefer to keep it as it is.